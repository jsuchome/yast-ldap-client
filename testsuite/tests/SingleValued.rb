# encoding: utf-8

#  SingleValued.ycp
#  Test of Ldap::SingleValued function
#  Author:	Jiri Suchomel <jsuchome@suse.cz>
#  $Id$
module Yast
  class SingleValuedClient < Client
    def main
      Yast.include self, "testsuite.rb"
      # testedfiles: Ldap.ycp

      @READ = {
        "target"  => { "size" => -1 },
        "product" => { "features" => { "EVMS_CONFIG" => "nazdar" } }
      }
      TESTSUITE_INIT([@READ, {}, {}], nil)

      Yast.import "Ldap"

      DUMP("==================================================")
      DUMP("===== single attribute ===========================")
      @READ1 = {
        "ldap" => {
          "schema" => { "at" => { "desc" => "uid desc", "single" => true } }
        }
      }

      TEST(lambda { Ldap.SingleValued("uid") }, [@READ1, {}, {}], 0)

      DUMP("==================================================")
      DUMP("===== non-existent attribute =====================")

      @READ2 = { "ldap" => { "schema" => { "at" => nil } } }

      TEST(lambda { Ldap.SingleValued("skeldi") }, [@READ2, {}, {}], 0)

      DUMP("==================================================")
      DUMP("===== non-single attribute =======================")

      @READ3 = {
        "ldap" => { "schema" => { "at" => { "desc" => "skelDir desc" } } }
      }
      TEST(lambda { Ldap.SingleValued("skeldir") }, [@READ3, {}, {}], 0)

      DUMP("==================================================")
      DUMP("===== skelDir once again (already in cache) ======")

      TEST(lambda { Ldap.SingleValued("skeldir") }, [@READ3, {}, {}], 0)

      DUMP("==================================================")
      DUMP("===== cn (not single) ============================")

      @READ4 = {
        "ldap" => {
          "schema" => {
            "at" => {
              "desc"   => "cn attribute description",
              "oid"    => "1.2.3.4.5",
              "single" => false
            }
          }
        }
      }

      TEST(lambda { Ldap.SingleValued("cn") }, [@READ4, {}, {}], 0)

      DUMP("==================================================")
      DUMP("===== description of new attribute ===============")

      @READ5 = {
        "ldap" => {
          "schema" => {
            "at" => {
              "desc"   => "The DN of a template that should be used by default",
              "single" => false
            }
          }
        }
      }

      TEST(lambda { Ldap.AttributeDescription("defaulttemplate") }, [
        @READ5,
        {},
        {}
      ], 0)
      DUMP("==================================================")
      DUMP("===== description of used attribute (in cache) ===")

      TEST(lambda { Ldap.AttributeDescription("cn") }, [@READ5, {}, {}], 0)

      nil
    end
  end
end

Yast::SingleValuedClient.new.main
