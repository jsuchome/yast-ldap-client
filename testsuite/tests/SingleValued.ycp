/**
 *  SingleValued.ycp
 *  Test of Ldap::SingleValued function
 *  Jiri Suchomel <jsuchome@suse.cz>
 *  $Id$
 */

{

    include "testsuite.ycp";
    import "Ldap";

    DUMP ("==================================================");
    DUMP ("===== single attribute ===========================");
    map READ1 =	$[ "ldap": $[
	"schema": $[
	    "at": $[
		"desc": "uid desc",
		"single": true
	    ]
	]
    ]];

    TEST(``(Ldap::SingleValued ("uid")), [ READ1, $[], $[] ], 0);

    DUMP ("==================================================");
    DUMP ("===== non-existent attribute =====================");

    map READ2 =	$[ "ldap": $[
	"schema": $[
	    "at": nil
	]
    ]];

    TEST(``(Ldap::SingleValued ("skelDi")), [ READ2, $[], $[] ], 0);

    DUMP ("==================================================");
    DUMP ("===== non-single attribute =======================");

    map READ3 =	$[ "ldap": $[
	"schema": $[
	    "at": $[
		"desc": "skelDir desc",
	    ]
	]
    ]];
    TEST(``(Ldap::SingleValued ("skelDir")), [ READ3, $[], $[] ], 0);

    DUMP ("==================================================");
    DUMP ("===== skelDir once again (already in cache) ======");

    TEST(``(Ldap::SingleValued ("skelDir")), [ READ3, $[], $[] ], 0);

    DUMP ("==================================================");
    DUMP ("===== cn (always single) =========================");

    map READ4 =	$[ "ldap": $[
	"schema": $[
	    "at": $[
		"desc": "cn attribute description",
		"oid": "1.2.3.4.5",
		"single": false
	    ]
	]
    ]];

    TEST(``(Ldap::SingleValued ("cn")), [ READ4, $[], $[] ], 0);

    DUMP ("==================================================");
    DUMP ("===== description of new attribute ===============");

    map READ5 =	$[ "ldap": $[
	"schema": $[
	    "at": $[
		"desc": "The DN of a template that should be used by default",
		"single": false
	    ]
	]
    ]];

    TEST(``(Ldap::AttributeDescription ("defaultTemplate")), [ READ5, $[], $[] ], 0);
    DUMP ("==================================================");
    DUMP ("===== description of used attribute (in cache) ===");

    TEST(``(Ldap::AttributeDescription ("cn")), [ READ5, $[], $[] ], 0);


}
