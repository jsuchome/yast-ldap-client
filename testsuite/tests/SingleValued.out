Dump	==================================================
Dump	===== single attribute ===========================
Read	.ldap.schema.at $["name":"uid"] $["desc":"uid desc", "single":true]
Return	true
Dump	==================================================
Dump	===== non-existent attribute =====================
Read	.ldap.schema.at $["name":"skelDi"] nil
Return	false
Dump	==================================================
Dump	===== non-single attribute =======================
Read	.ldap.schema.at $["name":"skelDir"] $["desc":"skelDir desc"]
Return	false
Dump	==================================================
Dump	===== skelDir once again (already in cache) ======
Return	false
Dump	==================================================
Dump	===== cn (always single) =========================
Read	.ldap.schema.at $["name":"cn"] $["desc":"cn attribute description", "oid":"1.2.3.4.5", "single":false]
Return	true
Dump	==================================================
Dump	===== description of new attribute ===============
Read	.ldap.schema.at $["name":"defaultTemplate"] $["desc":"The DN of a template that should be used by default", "single":false]
Return	The DN of a template that should be used by default
Dump	==================================================
Dump	===== description of used attribute (in cache) ===
Return	cn attribute description
